src/main/java/mate/academy/Main.java

```diff
@@ -2,11 +2,12 @@
 
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class Main {
     public static void main(String[] args) {
         // Feel free to play with AsyncRequestProcessor in this main method if you want
-        ExecutorService executor = null; // Provide implementation that fits your needs
+        ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
let's add a top cap to the number of the threads that will be created, its only a 4 element list that we are working with, so there is no need in too much threads
```suggestion
        ExecutorService executor = Executors.newCachedThreadPool(5);
```

Reply:
fix